src/Service/Configurations/RuntimeConfigValidator.cs

```diff
@@ -283,6 +328,120 @@ public static void ValidateEntityNamesInConfig(Dictionary<string, Entity> entity
+            {
+                // The rest path can't be null.
+                throw new DataApiBuilderException(
+                    message: $"Entity: {entityName} has a null rest {RestEntitySettings.PROPERTY_PATH}. Accepted value types are: string, boolean.",
+                    statusCode: HttpStatusCode.ServiceUnavailable,
+                    subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError
+                    );
+            }
+
+            if (restPathElement.ValueKind is not JsonValueKind.True && restPathElement.ValueKind is not JsonValueKind.False
+                && restPathElement.ValueKind is not JsonValueKind.String)
+            {
+                // The rest path can only be a string or a boolean value.
+                throw new DataApiBuilderException(
+                    message: $"Entity: {entityName} has rest {RestEntitySettings.PROPERTY_PATH} specified with incorrect data type. " +
+                    $"Accepted data types are: string, boolean.",
+                    statusCode: HttpStatusCode.ServiceUnavailable,
+                    subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError
+                    );
+            }
+
+            if (restPathElement.ValueKind is JsonValueKind.String)
+            {
+                string path = restPathElement.ToString().TrimStart('/').TrimStart(' ');
+                if (string.IsNullOrEmpty(path))
+                {
+                    // The rest 'path' cannot be empty.
+                    throw new DataApiBuilderException(
+                        message: $"Entity: {entityName} has an empty rest {RestEntitySettings.PROPERTY_PATH}.",
+                        statusCode: HttpStatusCode.ServiceUnavailable,
+                        subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError
+                        );
+                }
+
+                return path;
+            }
+
+            return entityName;
+        }
+
+        /// <summary>
+        /// Helper method to validate that the Rest methods are correctly configured for the entity.
+        /// Rest methods can only be configured for stored procedures as an array of valid REST operations.
+        /// </summary>
+        /// <param name="entityName">Name of the entity.</param>
+        /// <param name="restMethodsElement">Rest methods element configured for the entity.</param>
+        /// <param name="entity">Entity object.</param>
+        /// <exception cref="DataApiBuilderException">Throws exception whenever a the rest methods are configured for a non-stored procedure entity or
+        /// contain an unexpected value.</exception>
+        private static void ValidateRestMethodsForEntity(string entityName, JsonElement restMethodsElement, Entity entity)
+        {
+            // This is needed to correctly populate the source type for the entity.
+            entity.TryPopulateSourceFields();
```

Comment:
Shouldn't this population be done before calling this validation workflow? What happens if in Production mode since we wont do any of these validations, who will be calling TryPopulateSourceFields?

Reply:
This method is called by `ValidateEntityConfiguration` which is executed only when we are operating in development mode.